actors/builtin/storage_miner/storage_miner_actor_state.go

```diff
@@ -71,126 +102,150 @@ type SectorOnChainInfo struct {
 	}
+	return &info, found, nil
+}
 
-	ret = storage_power.SectorStorageWeightDesc{
-		SectorSize: st.Info.SectorSize,
-		DealWeight: sectorInfo.DealWeight,
-		Duration:   sectorInfo.Info.Expiration - sectorInfo.ActivationEpoch,
+func (st *StorageMinerActorState) deleteSector(store adt.Store, sectorNo abi.SectorNumber) error {
+	sectors := adt.AsArray(store, st.Sectors)
+	err := sectors.Delete(uint64(sectorNo))
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete sector %v", sectorNo)
 	}
-	ok = true
-	return
+	return nil
 }
 
-func (st *StorageMinerActorState) _getStorageWeightDescForSector(sectorNumber abi.SectorNumber) storage_power.SectorStorageWeightDesc {
-	ret, found := st.GetStorageWeightDescForSectorMaybe(sectorNumber)
-	Assert(found)
-	return ret
+func (st *StorageMinerActorState) forEachSector(store adt.Store, f func(*SectorOnChainInfo)) error {
+	sectors := adt.AsArray(store, st.Sectors)
+	var sector SectorOnChainInfo
+	err := sectors.ForEach(&sector, func(idx int64) error {
+		f(&sector)
+		return nil
+	})
+	return err
 }
 
-func (st *StorageMinerActorState) _getStorageWeightDescsForSectors(sectorNumbers []abi.SectorNumber) []storage_power.SectorStorageWeightDesc {
-	ret := []storage_power.SectorStorageWeightDesc{}
-	for _, sectorNumber := range sectorNumbers {
-		ret = append(ret, st._getStorageWeightDescForSector(sectorNumber))
+func (st *StorageMinerActorState) GetStorageWeightDescForSector(store adt.Store, sectorNo abi.SectorNumber) (*storage_power.SectorStorageWeightDesc, error) {
+	sectorInfo, found, err := st.getSector(store, sectorNo)
+	if err != nil {
+		return nil, err
+	} else if !found {
+		return nil, errors.Errorf("no such sector %v", sectorNo)
 	}
-	return ret
+
+	return asStorageWeightDesc(st.Info.SectorSize, sectorInfo), nil
 }
 
-func (st *StorageMinerActorState) IsSectorInTemporaryFault(sectorNumber abi.SectorNumber) bool {
-	checkSector, found := st.Sectors[sectorNumber]
-	if !found {
-		return false
+func (st *StorageMinerActorState) IsSectorInTemporaryFault(store adt.Store, sectorNo abi.SectorNumber) (bool, error) {
+	sectorInfo, found, err := st.getSector(store, sectorNo)
+	if err != nil {
+		return false, err
+	} else if !found {
+		return false, errors.Errorf("no such sector %v", sectorNo)
 	}
-	ret, err := st.FaultSet.Has(uint64(sectorNumber))
+	ret, err := st.FaultSet.Has(uint64(sectorNo))
 	AssertNoError(err)
-	Assert(checkSector.DeclaredFaultEpoch != epochUndefined)
-	Assert(checkSector.DeclaredFaultDuration != epochUndefined)
-	return ret
+	Assert(sectorInfo.DeclaredFaultEpoch != epochUndefined)
+	Assert(sectorInfo.DeclaredFaultDuration != epochUndefined)
+	return ret, nil
 }
 
-func (st *StorageMinerActorState) GetCurrentProvingSet() cid.Cid {
+func (st *StorageMinerActorState) ComputeProvingSet() cid.Cid {
 	// Current ProvingSet is a snapshot of the Sectors AMT subtracting sectors in the FaultSet
+	// TODO: actually implement this
```

Comment:
should this just panic? I don't think there is any calling code

Reply:
It is called.